In [2]:
import numpy as np

In [3]:
data=np.genfromtxt("heart.csv", delimiter=",", skip_header=True)

In [4]:
data

array([[63.,  1.,  3., ...,  0.,  1.,  1.],
       [37.,  1.,  2., ...,  0.,  2.,  1.],
       [41.,  0.,  1., ...,  0.,  2.,  1.],
       ...,
       [68.,  1.,  0., ...,  2.,  3.,  0.],
       [57.,  1.,  0., ...,  1.,  3.,  0.],
       [57.,  0.,  1., ...,  1.,  2.,  0.]])

In [7]:
age=data[:,0]
print(f"age: {age[:5]}")
sex=data[:,1]
print(f"sex: {sex[:5]}")
trestbps=data[:,3]
print(f"trestbps: {trestbps[:5]}")
chol=data[:,4]
print(f"chol: {chol[:5]}")
thalach=data[:,7]
print(f"thalach: {thalach[:5]}")
target=data[:,13]
print(f"target: {target[:5]}")

age: [63. 37. 41. 56. 57.]
sex: [1. 1. 0. 1. 0.]
trestbps: [145. 130. 130. 120. 120.]
chol: [233. 250. 204. 236. 354.]
thalach: [150. 187. 172. 178. 163.]
target: [1. 1. 1. 1. 1.]


### Basic descriptive statistics:

In [8]:
# Age
print("---Basic Descriptive Statistics on Age---")
print(f"Mean: {round(np.mean(age))}")
print(f"Median: {np.median(age)}")
print(f"Standard Deviation: {round(np.std(age))}")
print(f"Min: {np.min(age)}")
print(f"Max: {np.max(age)}")

---Basic Descriptive Statistics on Age---
Mean: 54
Median: 55.0
Standard Deviation: 9
Min: 29.0
Max: 77.0


In [9]:
# Sex
print("---Basic Descriptive Statistics on Sex---")
print(f"Mean: {round(np.mean(sex))}")
print(f"Median: {np.median(sex)}")
print(f"Standard Deviation: {round(np.std(sex))}")
print(f"Min: {np.min(sex)}")
print(f"Max: {np.max(sex)}")

---Basic Descriptive Statistics on Sex---
Mean: 1
Median: 1.0
Standard Deviation: 0
Min: 0.0
Max: 1.0


In [10]:
# Trestbps
print("---Basic Descriptive Statistics on Trestbps---")
print(f"Mean: {round(np.mean(trestbps))}")
print(f"Median: {np.median(trestbps)}")
print(f"Standard Deviation: {round(np.std(trestbps))}")
print(f"Min: {np.min(trestbps)}")
print(f"Max: {np.max(trestbps)}")

---Basic Descriptive Statistics on Trestbps---
Mean: 132
Median: 130.0
Standard Deviation: 18
Min: 94.0
Max: 200.0


In [11]:
# Chol
print("---Basic Descriptive Statistics on Chol---")
print(f"Mean: {round(np.mean(chol))}")
print(f"Median: {np.median(chol)}")
print(f"Standard Deviation: {round(np.std(chol))}")
print(f"Min: {np.min(chol)}")
print(f"Max: {np.max(chol)}")

---Basic Descriptive Statistics on Chol---
Mean: 246
Median: 240.0
Standard Deviation: 52
Min: 126.0
Max: 564.0


In [12]:
# Thalach
print("---Basic Descriptive Statistics on Thalach---")
print(f"Mean: {round(np.mean(thalach))}")
print(f"Median: {np.median(thalach)}")
print(f"Standard Deviation: {round(np.std(thalach))}")
print(f"Min: {np.min(thalach)}")
print(f"Max: {np.max(thalach)}")

---Basic Descriptive Statistics on Thalach---
Mean: 150
Median: 153.0
Standard Deviation: 23
Min: 71.0
Max: 202.0


In [13]:
# Target
print("---Basic Descriptive Statistics on Target---")
print(f"Mean: {round(np.mean(target))}")
print(f"Median: {np.median(target)}")
print(f"Standard Deviation: {round(np.std(target))}")
print(f"Min: {np.min(target)}")
print(f"Max: {np.max(target)}")

---Basic Descriptive Statistics on Target---
Mean: 1
Median: 1.0
Standard Deviation: 0
Min: 0.0
Max: 1.0


### Data Filtering

In [14]:
print(f"Number of rows before filtering: {len(data)}")

Number of rows before filtering: 303


In [31]:
filtered_data=data[(age>50) & (chol>250) & (target==1) & (sex==1)]
print(f"Number of rows after filtering: {len(filtered_data)}")

Number of rows after filtering: 12


### Number of unique categories

In [17]:
print(f"Unique trestbps counts: {np.unique(trestbps)}")
print(f" How many unique trestbps: {len(np.unique(trestbps))}")

Unique trestbps counts: [ 94. 100. 101. 102. 104. 105. 106. 108. 110. 112. 114. 115. 117. 118.
 120. 122. 123. 124. 125. 126. 128. 129. 130. 132. 134. 135. 136. 138.
 140. 142. 144. 145. 146. 148. 150. 152. 154. 155. 156. 160. 164. 165.
 170. 172. 174. 178. 180. 192. 200.]
 How many unique trestbps: 49


In [18]:
print(f"Unique chol counts: {np.unique(chol)}")
print(f" How many unique chol: {len(np.unique(chol))}")

Unique chol counts: [126. 131. 141. 149. 157. 160. 164. 166. 167. 168. 169. 172. 174. 175.
 176. 177. 178. 180. 182. 183. 184. 185. 186. 187. 188. 192. 193. 195.
 196. 197. 198. 199. 200. 201. 203. 204. 205. 206. 207. 208. 209. 210.
 211. 212. 213. 214. 215. 216. 217. 218. 219. 220. 221. 222. 223. 224.
 225. 226. 227. 228. 229. 230. 231. 232. 233. 234. 235. 236. 237. 239.
 240. 241. 242. 243. 244. 245. 246. 247. 248. 249. 250. 252. 253. 254.
 255. 256. 257. 258. 259. 260. 261. 262. 263. 264. 265. 266. 267. 268.
 269. 270. 271. 273. 274. 275. 276. 277. 278. 281. 282. 283. 284. 286.
 288. 289. 290. 293. 294. 295. 298. 299. 300. 302. 303. 304. 305. 306.
 307. 308. 309. 311. 313. 315. 318. 319. 321. 322. 325. 326. 327. 330.
 335. 340. 341. 342. 353. 354. 360. 394. 407. 409. 417. 564.]
 How many unique chol: 152


In [19]:
print(f"Unique thalach counts: {np.unique(thalach)}")
print(f" How many unique thalach: {len(np.unique(thalach))}")

Unique thalach counts: [ 71.  88.  90.  95.  96.  97.  99. 103. 105. 106. 108. 109. 111. 112.
 113. 114. 115. 116. 117. 118. 120. 121. 122. 123. 124. 125. 126. 127.
 128. 129. 130. 131. 132. 133. 134. 136. 137. 138. 139. 140. 141. 142.
 143. 144. 145. 146. 147. 148. 149. 150. 151. 152. 153. 154. 155. 156.
 157. 158. 159. 160. 161. 162. 163. 164. 165. 166. 167. 168. 169. 170.
 171. 172. 173. 174. 175. 177. 178. 179. 180. 181. 182. 184. 185. 186.
 187. 188. 190. 192. 194. 195. 202.]
 How many unique thalach: 91


### Correlation

In [26]:
corr_heartdisease=np.corrcoef(age,chol,rowvar=False)
print(f"Corr age and chol {corr_heartdisease}")

Corr age and chol [[1.         0.21367796]
 [0.21367796 1.        ]]


In [27]:
corr_heartdisease=np.corrcoef(trestbps,chol,rowvar=False)
print(f"Corr trestbps and chol {corr_heartdisease}")

Corr trestbps and chol [[1.         0.12317421]
 [0.12317421 1.        ]]


In [28]:
corr_heartdisease=np.corrcoef(thalach,chol,rowvar=False)
print(f"Corr thalach and chol {corr_heartdisease}")

Corr thalach and chol [[ 1.         -0.00993984]
 [-0.00993984  1.        ]]


### Analysis

* Age and Chol have a 21% which is weak positive
* Trestbps and Chol have a 12% which is weak positive
* Thalach and Chol have a -0.9% which is weak negative

### Summary

Initially, in our hypothesis we said that the older a male participant is and the higher their cholesteral levels, the more likely a patient is to get heart disease. After going through and testing the data, we realized that this was not the case and there is actually a completely different correlation between people getting heart disease. However, after analysing the difference in these statistics between male and female patients, we saw more of a correlation between the male patients that are older and have a higher cholesterol levels having heart disease, compared to the females that are older and have higher cholesterol levels